lib/Onyx.d.ts

```diff
@@ -0,0 +1,278 @@
+
+/**
+ * Keys added to this list can never be deleted.
+ *
+ * @param key
+ * @param connectionID
+ */
+declare function addToEvictionBlockList<TKey extends Key | CollectionKey>(key: TKey, connectionID: number): void;
+
+/**
+ * Subscribes a react component's state directly to a store key
+ *
+ * @example
+ * const connectionID = Onyx.connect({
+ *     key: ONYXKEYS.SESSION,
+ *     callback: onSessionChange,
+ * });
+ *
+ * @param mapping the mapping information to connect Onyx to the components state
+ * @param mapping.key ONYXKEY to subscribe to
+ * @param [mapping.statePropertyName] the name of the property in the state to connect the data to
+ * @param {Object} [mapping.withOnyxInstance] whose setState() method will be called with any changed data
+ *      This is used by React components to connect to Onyx
+ * @param [mapping.callback] a method that will be called with changed data
+ *      This is used by any non-React code to connect to Onyx
+ * @param [mapping.initWithStoredValues] If set to false, then no data will be prefilled into the
+ *  component
+ * @param [mapping.waitForCollectionCallback] If set to true, it will return the entire collection to the callback as a single object
+ * @param {String|Function} [mapping.selector] THIS PARAM IS ONLY USED WITH withOnyx(). If included, this will be used to subscribe to a subset of an Onyx key's data.
+ *       If the selector is a string, the selector is passed to lodashGet on the sourceData. If the selector is a function, the sourceData and withOnyx state are
+ *       passed to the selector and should return the simplified data. Using this setting on `withOnyx` can have very positive performance benefits because the component
+ *       will only re-render when the subset of data changes. Otherwise, any change of data on any property would normally cause the component to re-render (and that can
+ *       be expensive from a performance standpoint).
+ * @returns an ID to use when calling disconnect
+ */
+declare function connect<TKey extends Key | CollectionKey>(mapping: ConnectOptions<TKey>): number;
+
+/**
+ * Remove the listener for a react component
+ * @example
+ * Onyx.disconnect(connectionID);
+ *
+ * @param connectionID unique id returned by call to Onyx.connect()
+ * @param [keyToRemoveFromEvictionBlocklist]
+ */
+declare function disconnect<TKey extends Key | CollectionKey>(connectionID: number, keyToRemoveFromEvictionBlocklist?: TKey): void;
+
+/**
+ * Write a value to our store with the given key
+ *
+ * @param key ONYXKEY to set
+ * @param value value to store
+ */
+declare function set<TKey extends Key | CollectionKey>(key: TKey, value: Value[TKey] | null): Promise<void>;
+
+/**
+ * Sets multiple keys and values
+ *
+ * @example Onyx.multiSet({'key1': 'a', 'key2': 'b'});
+ *
+ * @param data object keyed by ONYXKEYS and the values to set
+ */
+declare function multiSet(data: Partial<KeyValueMap>): Promise<void>;
```

Comment:
Can new value for a key in data be `null`?

Reply:
@hayata-suenaga Good question, I need to test this.

Reply:
@hayata-suenaga It allows `null` values, will update it 👍